hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -181,6 +181,31 @@ public void incrementCount(List<Long> txIDs)
     }
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * @throws IOException
+   */
+  @Override
+  public int resetCount(List<Long> txIDs) throws IOException {
+    List<Long> failedTransactions = getFailedTransactions().stream()
+        .map(DeletedBlocksTransaction::getTxID).collect(Collectors.toList());
+    if (txIDs != null && !txIDs.isEmpty()) {
+      failedTransactions = failedTransactions.stream().filter(txIDs::contains)
+          .collect(Collectors.toList());
+    }
```

Comment:
Using set is a better choice, I would adopt it! But why the second stream is not allowed? These two streams are independent processes I think.  I  find no such convention, haha. Please let me know if it is true.

Reply:
> why the second stream is not allowed? These two streams are independent processes I think

It is allowed, but unnecessary.  Stream pipelines support multiple non-terminal operations (`filter()`, `map()`, etc.):

```suggestion
    Stream<Long> stream = getFailedTransactions().stream()
        .map(DeletedBlocksTransaction::getTxID);
    if (txIDs != null && !txIDs.isEmpty()) {
      stream = stream.filter(txIDs::contains);
    }
    List<Long> failedTransactions = stream.collect(Collectors.toList());
```

This is better than the one with `Set` if the filter is expected to drop many items.  We save the cost of materializing the bigger collection.

However, since this command is not performance-critical, both solutions are fine.

Reply:
Thx for the explanation!